pkg/operator/ceph/cluster/osd/status.go

```diff
@@ -218,10 +218,10 @@ func (c *Cluster) updateAndCreateOSDsLoop(
 		ResourceVersion: configMapList.ResourceVersion,
 	}
 	watcher, err := cmClient.Watch(c.clusterInfo.Context, watchOptions)
-	defer watcher.Stop()
 	if err != nil {
 		return false, errors.Wrapf(err, "failed to start watching OSD provisioning status ConfigMaps")
 	}
+	defer watcher.Stop()
```

Comment:
defer runs at very last I dont think so why it would be a prblem

Reply:
The problem is that the current code causes operator's crash altough the expected behavior is returning an error.

If `cmClient.Watch()` failed, `watcher` becomes `nil`.
If defer `watcher.Stop()` is set before the error check, this function is called just before running `L223`. It causes the operator crashes  due to null pointer dereference. The expected behavior is returnung error.

Reply:
@parth-gr Do you have any comment to my answer?

Reply:
Got it thanks for the explanation, we would be skipping the defer in such case

Reply:
Got it thanks for the explanation, we would be skipping the defer in such case